File: tests/fixtures/repos/js_sample/calculator.js
import { clamp } from './utils.js';

class Calculator {
    constructor() {
        this.history = [];
        this.result = 0;
    }

    add(n) {
        this.result += n;
        this.history.push({ op: 'add', n });
        return this;
    }

    subtract(n) {
        this.result -= n;
        this.history.push({ op: 'subtract', n });
        return this;
    }

    multiply(n) {
        if (n === 0) {
            this.result = 0;
        } else {
            this.result *= n;
        }
        this.history.push({ op: 'multiply', n });
        return this;
    }

    divide(n) {
        if (n === 0) {
            throw new Error('Division by zero');
        }
        this.result /= n;
        this.history.push({ op: 'divide', n });
        return this;
    }

    clampResult(min, max) {
        this.result = clamp(this.result, min, max);
        return this;
    }

    reset() {
        this.result = 0;
        this.history = [];
        return this;
    }

    getValue() {
        return this.result;
    }
}

export function evaluate(expression) {
    const tokens = expression.trim().split(/\s+/);
    let acc = parseFloat(tokens[0]);
    for (let i = 1; i < tokens.length - 1; i += 2) {
        const op = tokens[i];
        const operand = parseFloat(tokens[i + 1]);
        switch (op) {
            case '+': acc += operand; break;
            case '-': acc -= operand; break;
            case '*': acc *= operand; break;
            case '/':
                if (operand === 0) throw new Error('Division by zero');
                acc /= operand;
                break;
            default:
                throw new Error(`Unknown operator: ${op}`);
        }
    }
    return acc;
}

export default Calculator;

File: tests/fixtures/repos/py_sample/calculator.py
"""Simple calculator module for fixture testing."""

from __future__ import annotations


class Calculator:
    """A basic calculator with history."""

    def __init__(self) -> None:
        self.history: list[float] = []

    def add(self, a: float, b: float) -> float:
        result = a + b
        self.history.append(result)
        return result

    def subtract(self, a: float, b: float) -> float:
        result = a - b
        self.history.append(result)
        return result

    def multiply(self, a: float, b: float) -> float:
        result = a * b
        self.history.append(result)
        return result

    def divide(self, a: float, b: float) -> float:
        if b == 0:
            raise ValueError("Division by zero")
        result = a / b
        self.history.append(result)
        return result

    def clear_history(self) -> None:
        self.history.clear()


def parse_expression(expr: str) -> tuple[float, str, float]:
    """Parse a simple 'a op b' expression."""
    parts = expr.strip().split()
    if len(parts) != 3:
        raise ValueError(f"Invalid expression: {expr!r}")
    left, op, right = parts
    return float(left), op, float(right)


def evaluate(expr: str) -> float:
    """Evaluate a simple arithmetic expression."""
    calc = Calculator()
    left, op, right = parse_expression(expr)
    if op == "+":
        return calc.add(left, right)
    elif op == "-":
        return calc.subtract(left, right)
    elif op == "*":
        return calc.multiply(left, right)
    elif op == "/":
        return calc.divide(left, right)
    else:
        raise ValueError(f"Unknown operator: {op!r}")
